MimeKit/AttachmentCollection.cs

```diff
@@ -134,17 +135,20 @@ static void LoadContent (MimePart attachment, Stream stream)
 
 			if (attachment.ContentType.IsMimeType ("text", "*")) {
 				var filter = new BestEncodingFilter ();
-				var buf = new byte[4096];
+				int bufLength = 4096;
+				var buf = ArrayPool<byte>.Shared.Rent (bufLength);
 				int index, length;
 				int nread;
 
-				while ((nread = stream.Read (buf, 0, buf.Length)) > 0) {
+				while ((nread = stream.Read (buf, 0, bufLength)) > 0) {
 					filter.Filter (buf, 0, nread, out index, out length);
 					content.Write (buf, 0, nread);
 				}
 
 				filter.Flush (buf, 0, 0, out index, out length);
 
+				ArrayPool<byte>.Shared.Return (buf);
```

Comment:
I'd usually recommend wrapping the above code (while loop and flush) inside a try block and then putting the return inside a finally to ensure that it's returned despite potential exceptions. It'll get collected eventually but it's the pattern I've seen recommended.

Reply:
If you think about it, if there's an unhandled exception then it doesn't matter whether the buffer is returned or not since the application will break. It's all managed memory so the GC will collect the buffer eventually.

I could add the pattern though if that's what everyone wants.

Reply:
I must admit, I've not explored the usage of this and whether exceptions may be caught higher up. I'd assumed that might be the case and if so, there's a slight benefit to putting it back in the pool cleanly.

Reply:
I'll add the pattern then (I haven't explored either tbh.)